### ISSUE TITLE
Migration to Python 3.13

### ISSUE THREAD

**Issue by tarsil** [jcrist/msgspec#777]
### Question

@jcrist I know you are busy and thank you for everything you have done so far for all of us.

As you know, we have a lot of software depending on msgspec and we can't release until this is moved as well. WHa do you say to release the current 3.13 and later on you can release a new one for support of threading? This would unblock literally hundreds of people.

**Comment by ncoghlan** [comment]
Just noting for readers that haven't seen the previous threads that #711 is the relevant merged-but-not-released PR.

It can currently be obtained via a VCS install:

```
pip install git+https://github.com/jcrist/msgspec.git@main
```

`msgspec` correctly reports this style of installation by appending a local version to the version specifier:

$ pip install --user git+https://github.com/jcrist/msgspec.git@main
Collecting git+https://github.com/jcrist/msgspec.git@main
  ... snip build details ...
Successfully built msgspec
Installing collected packages: msgspec
Successfully installed msgspec-0.18.6+30.g595c33c
$ python3 -c "import msgspec; print(msgspec.__version__)"
0.18.6+30.g595c33c
$ python3 -c "from importlib.metadata import version; print(version('msgspec'))"
0.18.6+30.g595c33c

**Comment by ncoghlan** [comment]
As another interim workaround option, https://github.com/ncoghlan/msgspec/commits/py313-prerelease-tarball/ has a hardcoded release version that can be built from a GitHub source tarball:

```console
$ pip install --user "msgspec @ https://github.com/ncoghlan/msgspec/archive/a3bf004a48ac372394efa7378a1bc36d15dcf78d.zip"
Collecting msgspec@ https://github.com/ncoghlan/msgspec/archive/a3bf004a48ac372394efa7378a1bc36d15dcf78d.zip
    ... snip build details ...
Successfully built msgspec
Installing collected packages: msgspec
Successfully installed msgspec-0.18.7.dev1+ncoghlan.py313.prerelease
$ python3 -c "import msgspec; print(msgspec.__version__)"
0.18.7.dev1+ncoghlan-py313-prerelease
$ python3 -c "from importlib.metadata import version; print(version('msgspec'))"
0.18.7.dev1+ncoghlan.py313.prerelease
```

Diff at https://github.com/jcrist/msgspec/compare/main...ncoghlan:msgspec:py313-prerelease-tarball?expand=1 shows the only changes are to drop `versioneer` in favour of a hardcoded interim pre-release version number.

Let me know if there's anything I can do to help with the actual release (branches that need trying out, etc), but if there isn't, then I'll just offer my general thanks for the library and stick with the workaround until the full release is available.

**Comment by cofin** [comment]
FWIW, I've [published](https://pypi.org/project/litestar-msgspec/) a temporary build for the testing with Litestar.  It's designed to be a drop in replacement for when the final official Msgspec release is available.

You should be able to run `pip install litestar-msgspec` on any Python between 3.9 and 3.13.  [Here's the source](https://github.com/litestar-org/litestar-msgspec) in case anyone wants to inspect the build logs.

**Comment by jcrist** [comment]
msgspec 0.19.0 has been released with support for Python 3.13. The full changelog may be found [here](https://jcristharif.com/msgspec/changelog.html#version-0-19-0-2024-12-27).

Thank you everyone for your patience. Closing.
